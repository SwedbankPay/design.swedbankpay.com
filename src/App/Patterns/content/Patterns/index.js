import React from "react";
import { DocContainer } from "@docutils";
import { Link } from "react-router-dom";
import patternsRoutes from "@src/App/routes/patterns";

const Overview = () => (
	<section className="component-overview hide-arrow-icon">
		{patternsRoutes[0].routes.map(
			(card) =>
				card.icon && (
					<Link
						key={card.title}
						to={`../${card.path}`}
						className="cards cards-primary cards-wide"
					>
						<div className="cards-content flex-row align-items-center m-0 ">
							<i
								className={`material-icons${
									card.outlined ? " material-icons-outlined" : ""
								} mr-3`}
							>
								{card.icon}
							</i>
							<span className="h3 m-0">{card.title}</span>
						</div>
						<i className="material-icons material-icons-outlined">
							arrow_forward
						</i>
					</Link>
				)
		)}
	</section>
);

const Patterns = () => (
	<DocContainer>
		<p className="lead">
			Patterns are best practice solutions for common product features built up
			by combinations of various components.{" "}
		</p>
		<Overview />
	</DocContainer>
);

export default Patterns;

export { Overview };
