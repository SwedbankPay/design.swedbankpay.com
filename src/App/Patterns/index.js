import React from "react";

import { RenderPage } from "../utils";

const Patterns = () => (
    <RenderPage path="/patterns" initPath="/patterns/patterns" />
);

export default Patterns;
