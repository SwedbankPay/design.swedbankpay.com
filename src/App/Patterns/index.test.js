import React from "react";
import renderer from "react-test-renderer";
import Patterns from "./index";

describe("Patterns: index", () => {
    it("is defined", () => {
        expect(Patterns).toBeDefined();
    });

    it.skip("renders", () => {
        const patterns = renderer.create(<Patterns />);

        expect(patterns.toJSON()).toMatchSnapshot();
    });
});
