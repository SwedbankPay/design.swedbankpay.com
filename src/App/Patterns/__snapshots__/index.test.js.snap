// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patterns: index renders 1`] = `
<div
  className="doc-container"
>
  <span
    className="dg-current-version text-uppercase"
  >
    Design Guide â€“ v. 
    10.7.3
  </span>
</div>
`;
